docs/mgmt/readme.md

```diff
@@ -0,0 +1,660 @@
+# Management plane SDK generator - how it works and its corresponding configurations
+
+## Resource hierarchy in management plane SDK
+
+For the basic management plane SDK concepts, please refer the detailed document [here](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/resourcemanager/Azure.ResourceManager/README.md#key-concepts) first.
+
+A resource is usually represented by three classes in the generated SDK: a `[Resource]Data` class, a `[Resource]Resource` class, and a `[Resource]Collection` class.
+
+### `[Resource]Data` class
+
+This class represents the "model" of the resource, and it corresponds to a model definition in the swagger, usually with the same name by trimming the word `Data` off.
+
+### `[Resource]Resource` class
+
+This class represents the a full resource client object which contains a `Data` property exposing the details as a `[Resource]Data` type.
+
+### `[Resource]Collection` class
+
+This class contains operations for a resource collection that belongs to a specific parent resource. It provides most of the logical collection operations.
```

Comment:
This part is related to the object hierarchy design, consider to introduce a little bit